Cabal/Distribution/Simple/GHC.hs

```diff
@@ -1074,22 +1074,20 @@ exeMainModuleName Executable{buildInfo = bnfo} =
 -- https://github.com/haskell/cabal/pull/4539#discussion_r118981753.
 decodeMainIsArg :: String -> Maybe ModuleName
 decodeMainIsArg arg
-  | not (null main_fn) && isLower (head main_fn)
-                        -- The arg looked like "Foo.Bar.baz"
+  | any isLower (safeHead main_fn)
```

Comment:
this is use of `any` makes me think, than smile.

On the edge of over-cleverness.

I'd rather define a helper `headOf main_fn isLower` function.

Reply:
Perhaps.  I've found it works fine as "any head of main_fn isLower" or "the head of main_fn, if any, isLower".

I'll put a little helper to... help it :)

Reply:
My brain simplified that to `any isLower main_fn` which made me think *this cannot be right*. `safeHead` is not well known function, so it makes you think.